autopilot/contractor.go

```diff
@@ -292,13 +299,15 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) e
 	}
 
 	// run contract refreshes
-	refreshed, err := c.runContractRefreshes(ctx, w, &remaining, address, toRefresh)
+	refreshed, discarded, err := c.runContractRefreshes(ctx, w, &remaining, address, toRefresh)
 	if err != nil {
 		c.logger.Errorf("failed to refresh contracts, err: %v", err) // continue
 	}
 
-	// build the new contract set (excluding formed contracts)
+	// add discarded contracts to the ignore list
+	toIgnore = append(toIgnore, discarded...)
```

Comment:
Can you remind me why we need to explicitly track the contracts to ignore instead of just ignoring them and not passing them to `buildContractSet`?

Reply:
similar to <https://github.com/SiaFoundation/renterd/pull/263#discussion_r1158413437> so resolving this one